compiler/rustc_error_codes/src/error_codes/E0161.md

```diff
@@ -4,11 +4,18 @@ Erroneous code example:
 
```

Comment:
Why change this file?

Reply:
The previous example no longer works, since the "box expression type must be Sized" error is emitted before E0161.  This causes the following CI break: https://github.com/rust-lang/rust/runs/3342569972

Reply:
Can you move that to a separate PR? Along with any test changes. That way it's a bit more clear the direct impact of the change in diagnostics here.

Reply:
Or, if you want to switch the commit orders to have the test and example changes be a commit before and everything else after, that works too.

Reply:
Sure, I've split it up as follows:
1. All changes to E0161; this commit should compile by itself.
2. Code updates.
3. Update tests where the error message has changed, plus add a regression test for this issue specifically.